packages/client/src/hooks/useAccount.js

```diff
@@ -0,0 +1,85 @@
+import { mutate, query, tx } from "@onflow/fcl";
+import {
+  GET_USER_FLOW_BALANCE,
+  GET_USER_FUSD_BALANCE,
+  GET_USER_USDC_BALANCE,
+  SEND_TOKENS_TO_TREASURY,
+} from "flow";
+import { COIN_TYPE_TO_META } from "constants/maps";
+import { COIN_TYPES } from "constants/enums";
+import { REGULAR_LIMIT } from "constants/constants";
+
+const COIN_TYPE_LIST = [COIN_TYPES.FLOW, COIN_TYPES.FUSD, COIN_TYPES.USDC];
+
+export default function useAccount() {
+  const doSendTokensToTreasury = async (treasuryAddr, amount, coinType) => {
+    const vaultPath = COIN_TYPE_TO_META[coinType].storageVaultPath;
+    return await mutate({
+      cadence: SEND_TOKENS_TO_TREASURY,
+      args: (arg, t) => [
+        arg(treasuryAddr, t.Address),
+        arg(amount, t.UFix64),
+        arg(vaultPath, t.Path),
+      ],
+      limit: REGULAR_LIMIT,
+    });
+  };
+
+  //FOR TESTING UTILS ONLY, calling this method deposits tokens to treasury
+  //should only be used for testing
+  const initDepositTokensToTreasury = async (treasuryAddr) => {
+    for await (const coinType of COIN_TYPE_LIST) {
+      try {
+        const res = await doSendTokensToTreasury(
+          treasuryAddr,
+          String(parseFloat(10).toFixed(8)),
+          coinType
+        );
+        await tx(res).onceSealed();
+      } catch (err) {
+        console.log(`Failed to deposit ${coinType}, error: ${err}`);
+      }
+    }
+  };
+
+  const getBalanceByCoinType = async (coinType, address) => {
+    let cadence;
+    switch (coinType) {
+      case COIN_TYPES.FLOW:
+        cadence = GET_USER_FLOW_BALANCE;
+        break;
+      case COIN_TYPES.FUSD:
+        cadence = GET_USER_FUSD_BALANCE;
+        break;
+      case COIN_TYPES.USDC:
+        cadence = GET_USER_USDC_BALANCE;
+      default:
+        cadence = GET_USER_FLOW_BALANCE;
```

Comment:
+1

Reply:
The only reason I'm hesitate to do that is, for each token type we want to retrieve balance we will need to import the token's contract (since not only the path is different, the vault type is different as well) and add to the conditions inside of the script (imagine moving the above switch statement into the script), that means this query script will get modified a lot. Feels like an anti-pattern for me. Not sure if we have a way to dynamically import contract in the script level?

Reply:
i was thinking something like this: https://github.com/DapperCollectives/VESSEL/blob/c13812bd4c508cb97cf9323ca9e4453b4087b422/packages/client/src/flow/addCollection.tx.js#L1-L5, we can use token to fetch it's corresponding contractName and path by utilizing maps.js then pass into the template script
https://github.com/DapperCollectives/VESSEL/blob/c13812bd4c508cb97cf9323ca9e4453b4087b422/packages/client/src/constants/maps.js#L12-L17

Reply:
Sounds goood, I'll give it a try!

Reply:
@amyliu6  @jacksonConrad Implemented the above solution, one caveat I would love to mention is this solution might break when the publicBalanceVault is configurable (for example USDC) as the publicBalanceVault might be different than what we defined in the maps.js. Assuming the configuration of USDC is not going to change frequently, this solution should work fine on testnet and mainnet

Reply:
Awesome! I would not expect the public path to change, in general... it would be a lot of overhead for any apps using USDC (like ours) so they are definitely incentivized not to change it.